File: apps/gig-crawler-2/src/models/gig.ts
import { z } from "zod";

export const GigSchema = z.object({
  title: z.string(),
  date: z.date(),
  venueName: z.string(),
  description: z.string().optional(),
  price: z.string().optional(),
  url: z.string().optional(),
  imageUrl: z.string().optional(),
});

export type Gig = z.infer<typeof GigSchema>;

export const StrapiGigSchema = z.object({
  data: z.object({
    title: z.string(),
    date: z.string(), // ISO string for Strapi
    venue: z.number(), // venue ID
    description: z.string().optional(),
    price: z.string().optional(),
    url: z.string().optional(),
    // imageUrl removed - Strapi schema doesn't support it
  }),
});

export type StrapiGig = z.infer<typeof StrapiGigSchema>;

export function toStrapiGig(gig: Gig, venueId: number): StrapiGig {
  return {
    data: {
      title: gig.title,
      date: gig.date.toISOString(),
      venue: venueId,
      description: gig.description,
      price: gig.price,
      url: gig.url,
      // imageUrl intentionally excluded - Strapi schema doesn't support it
    },
  };
}

File: apps/gig-crawler-2-python-backup/src/models/gig.py
"""Gig models matching Strapi schema."""
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field


class Gig(BaseModel):
    """Gig model for business logic."""
    title: str
    date: datetime
    venue_name: str
    description: Optional[str] = None
    price: Optional[str] = None
    url: Optional[str] = None
    image_url: Optional[str] = None


class StrapiGig(BaseModel):
    """Gig model for Strapi API."""
    data: dict = Field(...)

    @classmethod
    def from_gig(cls, gig: Gig, venue_id: int) -> "StrapiGig":
        """Convert Gig to Strapi format."""
        return cls(
            data={
                "title": gig.title,
                "date": gig.date.isoformat(),
                "venue": venue_id,
                "description": gig.description,
                "price": gig.price,
                "url": gig.url,
                "imageUrl": gig.image_url,
            }
        )
